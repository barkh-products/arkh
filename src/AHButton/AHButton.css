@media print {
    :local(.button) {
        display: none !important;
    }
}

:local(.button) {
    display: flex;
    flex: 0 1 auto;
    flex-direction: column;
    align-items: center;
    justify-content: center;
    flex-direction: row;
    color: #000;
    letter-spacing: 0.15em;
    margin: 5px 5%;
    padding: 1.2em;
    text-align: center;
    cursor: pointer;
    transition: opacity 0.2s;
    user-select: none;
    box-shadow: 0 0 0px 0 rgba(0, 0, 0, .008);
    font-weight: 400;
    border-radius: 0px;
}

:local(.button:hover) {
    box-shadow: 0 0 8px 0px rgba(0, 0, 0, 0.2);
    opacity: 0.7;
}

:local(.buttonLabel) {
    cursor: pointer;
    user-select: none;
}

:local(.disabled) {
    opacity: 0.7;
}

.loading {}