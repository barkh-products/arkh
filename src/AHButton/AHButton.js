// @flow
import React, { Children, PureComponent } from "react";
import styles from "./AHButton.css";

import { withTheme, AHThemeType } from "../AHTheme/AHTheme";
import { AHShakeableClickArea } from "../AHShakeableClickArea/AHShakeableClickArea";

type AHButtonTypes = "NORMAL" | "NEGATIVE" | "POSITIVE";

const typeStyles = (type: AHButtonTypes, theme: AHThemeType) =>
  ({
    NORMAL: {
      background: "#EEEEEE",
      color: theme.darkText
    },
    IMPORTANT: {
      background: theme.washedText,
      color: "#fff"
    },
    NEGATIVE: {
      background: theme.negativeColor,
      color: "#fff"
    },
    POSITIVE: {
      background: theme.positiveColor,
      color: "#fff"
    }
  }[type]);

class AHButtonUnthemed extends PureComponent {
  
  props: {
    children?: Children,
    type: AHButtonTypes,
    theme: AHThemeType,
    onClick: Event => void,
    style?: any,
    disabled: boolean,
    loading: boolean
  };

  static defaultProps = {
    type: "NORMAL",
    onClick: () => {}
  };

  displayName = "AHButton";

  renderLoading = () => <i className="fa fa-spinner fa-spin fa-1x fa-fw" />;

  render() {
    return (
      <div
        onClick={
          this.props.disabled || this.props.loading
            ? () => {}
            : this.props.onClick
        }
        style={{
          ...typeStyles(this.props.type, this.props.theme),
          ...this.props.style
        }}
        className={
          styles.button +
          " " +
          (this.props.disabled ? styles.disabled : "") +
          " " +
          (this.props.loading ? styles.loading : "")
        }
      >
        <AHShakeableClickArea shouldShake={this.props.disabled}>
          <label className={styles.buttonLabel}>
            {this.props.loading ? this.renderLoading() : this.props.children}
          </label>
        </AHShakeableClickArea>
      </div>
    );
  }
}

export const AHButton = withTheme(AHButtonUnthemed);

