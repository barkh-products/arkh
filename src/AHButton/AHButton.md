#### Normal Button

```js
<AHButton>A Regular Button</AHButton>
```

#### Negative Button

```js
<AHButton type="NEGATIVE">Push me</AHButton>
```

#### Positive Button

```js
<AHButton type="POSITIVE">Push me</AHButton>
```

#### Important Button

```js
<AHButton type="IMPORTANT">Push me</AHButton>
```

#### Submit Button

```js
<AHButton type="SUBMIT" />
```

### Disabled Buttons

```js
<AHButton>Push me</AHButton>
<AHButton type="NEGATIVE" disabled>Push me</AHButton>
<AHButton type="POSITIVE" disabled>Push me</AHButton>
<AHButton type="IMPORTANT" disabled>Push me</AHButton>
```
