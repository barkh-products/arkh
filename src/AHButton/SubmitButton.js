import React from "react";
import styled from "styled-components";

const Wrapper = styled.div`
  position: relative;
  :after {
    content: "";
    position: relative;
    background: #606060;
    display: block;
    height: 3px;
    width: 100%;
    transform: scaleX(0);
    transform-origin: 0 0;
    transition: 0.3s;
  }
  :hover {
    :after {
      transform: scaleX(1);
    }
  }
`;

const Submit = styled.input`
  margin-right: 0;
  border: none;
  outline: none;
  background: none;
  text-align: left;
  font-weight: 400;
  font-size: 17px;
  padding: 5px 0;
  cursor: pointer;
  :hover {
    border-bottom: 3px solid 606060;
  }
`;
export const SubmitButton = props => {
  const { loading, style, children, ...rest } = props;
  return (
    <Wrapper style={style}>
      {loading ? (
        <i className="fa fa-spinner fa-spin fa-2x fa-fw" />
      ) : (
        <Submit type="submit" {...rest} />
      )}
    </Wrapper>
  );
};