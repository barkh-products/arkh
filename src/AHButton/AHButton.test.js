/* @flow */
import React from "react";
import {shallow} from 'enzyme';
import renderer from "react-test-renderer";
import ReactTestUtils from 'react-dom/test-utils'; 
import styles from './AHButton.css';
import { AHButton } from "./AHButton";

describe("AHButton", () => {
  it("renders correctly", () => {
    const tree = renderer.create(<AHButton />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("should render children", () => {
    const tree = renderer.create(<AHButton ><div /></AHButton>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should render loading", () => {
    const tree = renderer.create(<AHButton loading={true} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should handle disabled", () => {
    const tree = renderer.create(<AHButton disabled={true} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

});
