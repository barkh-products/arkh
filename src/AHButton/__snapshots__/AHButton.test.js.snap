// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AHButton renders correctly 1`] = `
<div
  className="undefined  "
  onClick={[Function]}
  style={
    Object {
      "background": "#EEEEEE",
      "color": "black",
    }
  }
>
  <div
    className="sc-bdVaJa dqIiLf"
    onClick={[Function]}
  >
    <label
      className={undefined}
    />
  </div>
</div>
`;

exports[`AHButton should handle disabled 1`] = `
<div
  className="undefined undefined "
  onClick={[Function]}
  style={
    Object {
      "background": "#EEEEEE",
      "color": "black",
    }
  }
>
  <div
    className="sc-bdVaJa dqIiLf"
    onClick={[Function]}
  >
    <label
      className={undefined}
    />
  </div>
</div>
`;

exports[`AHButton should render children 1`] = `
<div
  className="undefined  "
  onClick={[Function]}
  style={
    Object {
      "background": "#EEEEEE",
      "color": "black",
    }
  }
>
  <div
    className="sc-bdVaJa dqIiLf"
    onClick={[Function]}
  >
    <label
      className={undefined}
    >
      <div />
    </label>
  </div>
</div>
`;

exports[`AHButton should render loading 1`] = `
<div
  className="undefined  undefined"
  onClick={[Function]}
  style={
    Object {
      "background": "#EEEEEE",
      "color": "black",
    }
  }
>
  <div
    className="sc-bdVaJa dqIiLf"
    onClick={[Function]}
  >
    <label
      className={undefined}
    >
      <i
        className="fa fa-spinner fa-spin fa-1x fa-fw"
      />
    </label>
  </div>
</div>
`;
